src/puppetlabs/pcp/client.clj

```diff
@@ -142,10 +142,13 @@
   "extracts the common name from the named certificate and forms a PCP
   Uri with it and the supplied type"
   [certificate type]
-  (let [x509     (ssl-utils/pem->cert certificate)
-        cn       (ssl-utils/get-cn-from-x509-certificate x509)
-        identity (format "pcp://%s/%s" cn type)]
-    identity))
+  (let [x509-chain (ssl-utils/pem->certs certificate)]
+    (when (empty? x509-chain)
+      (throw (IllegalArgumentException.
+               (i18n/trs "{0} must contain at least 1 certificate" certificate))))
+    (let [cn       (ssl-utils/get-cn-from-x509-certificate (first x509-chain))
```

Comment:
What do you say about adding a couple of lines in the `connect` docstring (last function, on the bottom) about our assumption on the certs bundle?